project/assets/asset_source_file.ts

```diff
@@ -324,18 +324,34 @@ Second path: ${chalk.yellow(sf.fsPath)}`,
   ): Promise<void> {
     const oldAutoloadClassName = this.getAutoloadNameFromExportedVariable()
 
+    let fsContent = await fs.readFile(this.fsPath, "utf-8")
     let sourceFileAst = watchProgram.getProgram().getSourceFile(this.fsPath)
     let tries = 0
 
-    while (!sourceFileAst && ++tries < 50) {
+    while (
+      // we always need the AST
+      (!sourceFileAst ||
+        // if we are in watch mode, as chokidar uses other functionality compared to TS to monitor files,
+        // sometimes we can race ahead of the TS compiler.
+        // Thus wait for them catch up with us.
+        (!this.project.args.buildOnly &&
+          fsContent !== sourceFileAst.getFullText())) &&
+      // if we don't have AST or are not synced, something went wrong
+      ++tries < 50
+    ) {
+      // wait 10ms
```

Comment:
nit: these comments are self-explanatory - probably unnecessary.

Reply:
I tend to be a bit more literal in code, but that is indeed out of fashion.